### PR TITLE
Switch stashing thread context back to only around system index calls

### PR DESCRIPTION
**Pull request by jmazanec15**
### Description

TLDR: Fix one wrapped stashed call for #827. Reverts #847 

In order for model index to work as a system index with security enabled, calls to modify the index need to be wrapped with the stashed thread context call. This will clear the headers of the thread context and allow the call to have permissions to modify the system index. In #827, we wrapped a few calls to the system index with the stashing of the thread context. However, the secure integ tests were failing due to the thread context being stashed around the wrong operation ([here](https://github.com/opensearch-project/k-NN/pull/827/files#diff-6e0a3411bcedc09941f45dc0f73c8b590700ca105020ae8c35dbebb8f4bdd857R307) instead of [here](https://github.com/opensearch-project/k-NN/pull/827/files#diff-6e0a3411bcedc09941f45dc0f73c8b590700ca105020ae8c35dbebb8f4bdd857R351)).

To get around this issue, the stashing of thread contexts were moved to the transport actions in #847. This allowed security integ tests to pass. However, this leads to a larger surface area of code executing with escalated privileges. For this reason, I think it makes sense to revert back to just wrapping calls to the system index with the stashed thread context.

 
### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by navneet1v** [comment]
@jmazanec15 thanks for taking care of this. Can we cut a github issue to refractor this code, where we provide 1 more abstraction layer on all our system index rather than ModelDao. The reason for this is, right now in ModelDao whenever we want to do some operation on the index we need to first stash the thread context. 
One abstraction can be and not limited to this:
We can use interceptors here, which can intercept the calls from K-NN to all the indices we are doing, validate if it is system index and add/remove appropriate information in thread context or any other place.


Doing that will allow seamless integration in other parts of K-NN and will also help us expand in future if we create more system indices.

**Review by navneet1v — APPROVED** [review]
approving the change. Please make sure that we create a github issue to refractor the ModelDao class.

**Review by naveentatikonda — APPROVED** [review]

**Comment by jmazanec15** [comment]
Will create an issue. We can probably discuss on options over there.
